### PR TITLE
Sync fief coop framework

### LINKED ISSUE #245: Fief Sync
Town interactions need to be synchronized via RPC and value persistence. Look into what other sync might be needed and implement.

The following fields will need to be synchronized
- [ ] FoodStocks

See https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/Coop/Mod/Persistence/Party/EventPartyMoveTo.cs. We use Railgun for RPC, https://github.com/araex/RailgunNet/blob/master/RailgunNet/Logic/RailEvent.cs

### PR DESCRIPTION
**Pull request by garrettluskey**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behaviour?
CoopFramework utilizes the MBObjectManager to hook object creation. This no longer works as most of that functionality has been moved to the CampaignObjectManager.

## What is the new behaviour?
The CoopFramework now only hooks constuctors.
Resolves #245 

## Other information

### CODE REVIEW

**Comment by thomasDelaporte** [comment]
The synchronization of the "foodStock" seems to work well.

+ I don't know if it's relevant to change this property in real time, see if we can optimize the bandwidth in the next version.

**I was able to try the branch very quickly, I think that the modifications made have changed the behavior for the time synchronization (it does not work anymore in the client) so the movements are impacted.**

File: source/Coop/Mod/GameSync/Fief/FiefCommands.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        //        if (fiefGuid == Guid.Empty)
+        //        {
+        //            return;
+        //        }
+
+        //        CoopServer.Instance.Synchronization.Broadcast(OnChangeFoodStockRPC.Id, null, new object[] { fiefGuid, fief.FoodStocks });
+        //    }
+        //}
+    }
```
Comment (thomasDelaporte) [inline source/Coop/Mod/GameSync/Fief/FiefCommands.cs]: I think you can remove this, it's the old RPC sync way.

File: source/Coop/Mod/GameSync/Fief/FiefCommands.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+using TaleWorlds.Library;
+
+namespace Coop.Mod.GameSync
+{
+    class FiefCommands
+    {
+        private const string sGroupName = "coop";
+        private const string sTestGroupName = "test";
+
+        [CommandLineFunctionality.CommandLineArgumentFunction("set_food_stocks", sTestGroupName)]
```
Comment (thomasDelaporte) [inline source/Coop/Mod/GameSync/Fief/FiefCommands.cs:30]: We could document these methods.

**Review by thomasDelaporte — CHANGES_REQUESTED** [review]
If you can check the problem I mentioned in my previous comment and clean up quickly and if possible document some methods if you think to keep them for the next project.

**Great job.**

File: source/Coop/Mod/GameSync/Fief/FiefCommands.cs (hunk 2 above)
Comment (garrettluskey) [reply source/Coop/Mod/GameSync/Fief/FiefCommands.cs:30]: Created https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Test-Game-Commands

**Comment by garrettluskey** [comment]
Time [issue](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/issues/254) was created. We can test and fix time and possibly party sync in this coming sprint.

**Review by thomasDelaporte — APPROVED** [review]

**Pull request merged**